### PR TITLE
Fix 500 when repo has invalid .editorconfig

### LINKED ISSUE #3643: Invalid data in .editorconfig will result in 500 error page
- Gogs version (or commit ref): 0.9.99.0903 (gogs/gogs docker)
- Git version: ??? (gogs/gogs docker)
- Operating system: Ubuntu 16.04 host (gogs/gogs docker)
- Database:
  - [x] PostgreSQL
  - [ ] MySQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gogs.io:
  - [x] Yes: https://try.gogs.io/bugmenot12/editorconfig-test
  - [ ] No
  - [ ] Not relevant
- Log gist:

```
[Macaron] 2016-09-13 08:41:45: Started GET /jvandertil/Lunar for ########
2016/09/13 08:41:45 [D] Session ID: 9d14413.......
2016/09/13 08:41:45 [D] CSRF Token: XmRgOBjxjPfbeQpuPZvN0eZvAzU......
2016/09/13 08:41:45 [...routers/repo/view.go:250 Home()] [E] Repo.GetEditorconfig: key-value delimiter not found: 0Zoeken

2016/09/13 08:41:45 [D] Template: status/500
```
## Description

Having invalid content in the .editorconfig file will result in a error 500 page. In the above example I had the text '0Zoeken' as the last line in the editorconfig file (don't know how it got there). This results in the 500 error page being displayed.

### PR DESCRIPTION
**Pull request by andreynering**
Creating a notice instead

Fixes gogits/gogs#3643

### CODE REVIEW

File: routers/repo/pull.go [hunk 1]
```diff
@@ -367,12 +367,10 @@ func ViewPullFiles(ctx *context.Context) {
 		return
 	}
 
-	ec, err := ctx.Repo.GetEditorconfig()
-	if err != nil && !git.IsErrNotExist(err) {
-		ctx.Handle(500, "ErrGettingEditorconfig", err)
+	setEditorconfigIfExists(ctx)
```
Comment (bkcsoft) [inline routers/repo/pull.go:371]: Shouldn't this be in the router instead of in the route?

File: routers/repo/pull.go [hunk 2]
```diff
@@ -626,12 +624,10 @@ func CompareAndPullRequest(ctx *context.Context) {
 		}
 	}
 
-	ec, err := ctx.Repo.GetEditorconfig()
-	if err != nil && !git.IsErrNotExist(err) {
-		ctx.Handle(500, "ErrGettingEditorconfig", err)
+	setEditorconfigIfExists(ctx)
```
Comment (bkcsoft) [inline routers/repo/pull.go:371]: This as well?

File: routers/repo/view.go [hunk 3]
```diff
@@ -245,12 +245,10 @@ func Home(ctx *context.Context) {
 		return
 	}
 
-	ec, err := ctx.Repo.GetEditorconfig()
-	if err != nil && !git.IsErrNotExist(err) {
-		ctx.Handle(500, "Repo.GetEditorconfig", err)
+	setEditorconfigIfExists(ctx)
```
Comment (bkcsoft) [inline routers/repo/view.go:248]: And this

File: routers/repo/commit.go [hunk 4]
```diff
@@ -179,12 +179,10 @@ func Diff(ctx *context.Context) {
 		}
 	}
 
-	ec, err := ctx.Repo.GetEditorconfig()
-	if err != nil && !git.IsErrNotExist(err) {
-		ctx.Handle(500, "ErrGettingEditorconfig", err)
+	setEditorconfigIfExists(ctx)
```
Comment (bkcsoft) [inline routers/repo/commit.go:182]: And here :slightly_smiling_face:

**Review by bkcsoft — COMMENTED** [review]
Maybe use `setEditorconfigIfExists` as a middleware instead of a function?

**Comment by strk** [comment]
Any chance you could add a test for this ?
I've tried this locally and confirm the bug and the fix, but I hadn't found where the notice about broken .editorconfig was sent. Where should I look ?

Other than that, LGTM

**Comment by andreynering** [comment]
@strk 

See the original issue https://github.com/gogits/gogs/issues/3643

To add a test, we should first have a mock of the git interface, or a real git repo, since this touch git. It's something for the future.

Also, how and why did you add two commits to this PR?

**Comment by strk** [comment]
I've added two commits by clicking "Update branch" button in github interface. I belive another button will let me rebase/squash before merge. Will try right now, as this LGTM (ok for postponing test)

**Comment by strk** [comment]
Uhm, it takes a second LGTM, mine was already counted

**Comment by tboerger** [comment]
Needs a rebase.

**Comment by strk** [comment]
Rebase can be done via github button, as long as LGTM passes

**Comment by tboerger** [comment]
> Rebase can be done via github button, as long as LGTM passes

Nope, there is no button to update, so maybe there are merge conflicts.

**Comment by strk** [comment]
Yup, looks like being the case, @andreynering can you manually rebase and resolve conflicts (then I guess it'll need another LGTM round ?)

**Comment by strk** [comment]
Merging  manually shows no conflicts, should I try pushing the merge via commandline ?

**Comment by tboerger** [comment]
The branch is out of sync, that's why this can only be done by admins, but we should avoid that.

**Comment by strk** [comment]
I remember having seen an "Update branch" button in other PRs, when they were out of sync, did any configuration change about that ? (or maybe the branch owner had the branch protected..)

**Pull request merged**